src/liballoc/boxed.rs

```diff
@@ -248,8 +250,9 @@ impl BoxAny for Box<Any> {
         if self.is::<T>() {
             unsafe {
                 // Get the raw representation of the trait object
+                let raw = into_raw(self);
                 let to: TraitObject =
-                    mem::transmute::<Box<Any>, TraitObject>(self);
+                    mem::transmute::<*mut Any, TraitObject>(raw);
```

Comment:
> I think transmuting from `Box<Any>` is fine here, adding the extra indirection is slightly confusing, as it makes the ownership transfer less clear.

IMO, transmuting `Box<T>` should not be used anywhere, because it makes code dependent on box implementation details.

> It seems that the `mem::transmute` on the next line could become `Ok(Box::from_raw(to.data as *mut T))`, though.

Good point.